README.md

```diff
@@ -56,4 +58,86 @@ as a template:
 
 ## CI Builds
 
-Building and pushing from [VSTS](https://devdiv.visualstudio.com/DevDiv/XamarinVS/_build/index?context=allDefinitions&path=%5CXamarin&definitionId=7445&_a=completed).
\ No newline at end of file
+Building and pushing from [VSTS](https://devdiv.visualstudio.com/DevDiv/XamarinVS/_build/index?context=allDefinitions&path=%5CXamarin&definitionId=7445&_a=completed).
+
+
+## Background Task
+
+If you want to start a background task and continue with the build
+and then wait for that background task later, you will need to use
+the `BackgroundTaskManager`. This class allows you to `Register` a
+Task object which will continue to run in the background. You can 
+then use the `WaitForBackgroundTasks` MSBuild Task  to hold the 
+build until that background task is complete. 
+
+```
+public class MyAsyncTask : Xamarin.Build.AsyncTask
+{
+  public override bool Execute()
+  {
+    var manager = BackgroundTaskManager.GetTaskManager (BuildEngine4);
+    var task = System.Threading.Tasks.Task.Run(async () =>
+    {
+      await System.Threading.Tasks.Task.Delay(10000);
+    });
+    manager.RegisterTask (task, "mycategory");
+
+    // DONT call base.Execute (); as we want to return immediately.
```

Comment:
Then this class should not even inherit from `AsyncTask`, should it? 
I mean, a regular task that schedules work with the background task 
manager, doesn't really need to do anything itself, does it?

It's sort of confusing, unless we offer a way for the task to register 
*itself* with the background manager to complete at a later time...

Reply:
It doesn't need do.